### DISCUSSION TITLE
Mithril Snapshot Storage

### DISCUSSION THREAD

**Discussion by Tarrant64** [input-output-hk/mithril#1514] [Ideas]
There seems to be some discussions over where snapshots are stored (currently google storage). How easy is this to change and/or distribute across other storage platforms - notably more decentralized ones if that's what users wanted?  Iagon has been brought up as a suggestion, but I don't know the technical reasons as to why google storage was used over other solutions so perhaps we should start there.

https://twitter.com/matthewjura/status/1759666356310487226

![image](https://github.com/input-output-hk/mithril/assets/126993855/ad4dae15-8c78-4e37-b6e1-5f3d2aa06928)

**Comment by jpraynaud** [comment]
Hi @Tarrant64, thanks for starting this discussion.

During the initial development phase of Mithril, we have made the tactical choice to work with a centralized CDN network.
This has allowed use to deploy the protocol swiftly to the Cardano `mainnet` in its current **beta** version.

However, since its inception, we have in mind that the Mithril aggregator should support various adapters to store the snapshot archives (whether on IPFS, BitTorrent, or another CDN). The issue with working on IPFS e.g. is that the snapshot archives that we store are currently `>40GB` which makes the download experience not reliable from what we have witnessed.

It will be possible to leverage decentralized storage for the `mainnet` in a more efficient way when the incremental snapshot signature is released (expected late 2024). In the mean time, we will probably start experimenting with IPFS for snapshots on the `testnets` which are much more slim.

Also, an important fact to note is that with Mithril we don't need to trust the provider of the snapshot as the certification process is based on verifying a multi-signature of the content of the snapshot archive itself.
